tests/test_read_write/test_read_csv.py

```diff
@@ -2,37 +2,85 @@
 from clumper import Clumper
 
 
-def test_read_csv():
-    "Test that the length of clumper matches the total number of rows in the csv."
-    clump = Clumper.read_csv("tests/iris.csv")
-    assert len(clump) == 150
-
-
-def test_read_csv_url():
-    "Test that data is read correctly from url, with the right number of rows."
-    clump = Clumper.read_csv("https://calmcode.io/datasets/monopoly.csv")
-    assert len(clump) == 22
-
-
-def test_read_csv_nrows():
-    "Test that the length of clumper matches the number of rows requested from the csv."
-    clump = Clumper.read_csv("tests/iris.csv", nrows=50)
-    assert len(clump) == 50
-
-
-def test_read_csv_nrows_url():
-    "Test that the length of clumper matches the number of rows requested from the url."
-    clump = Clumper.read_csv("https://calmcode.io/datasets/monopoly.csv", nrows=10)
-    assert len(clump) == 10
+@pytest.mark.parametrize(
```

Comment:
I like that parametric is added but there's a cleanup possible. 

1. Since the test always assumes `,` to be the delimiter. Why even add it in this list of settings? 
2. A unit test, even a parametrised one, is easier to reason about if it applies focus on one aspect. The first unit test might be "make sure that we get the appropriate number of rows". We don't need to also check the fieldnames in the same test, that can be done in another one. 
3.  I think it's fair to parametric the file source but maybe we can clean up some things by using `itertools.product`. 

Something like;

```python
import itertools as it
combinations = it.product(['path1', 'path2'], [(None, 22), (1, 1), (2, 2), (10, 10), (100, 22)])

@pytest.mark.parametrize("path, rows, expected", [(p, row, exp) for p, (row, exp) in prod])
def test_nrow_handles_fine(path, rows, expected):
    ...
```